programs/mango-v4/src/health/account_retriever.rs

```diff
@@ -99,11 +128,25 @@ pub fn new_fixed_order_account_retriever<'a, 'info>(
 }
 
 impl<T: KeyedAccountReader> FixedOrderAccountRetriever<T> {
-    fn bank(&self, group: &Pubkey, account_index: usize, token_index: TokenIndex) -> Result<&Bank> {
-        let bank = self.ais[account_index].load::<Bank>()?;
-        require_keys_eq!(bank.group, *group);
-        require_eq!(bank.token_index, token_index);
-        Ok(bank)
+    fn bank(
+        &self,
+        group: &Pubkey,
+        account_index: usize,
+        token_index: TokenIndex,
+    ) -> Result<(usize, &Bank)> {
+        for i in (0..=account_index).rev() {
+            let ai = &self.ais[i];
+            let bank = ai.load_fully_unchecked::<Bank>()?;
```

Comment:
this is a bug: the normal fixed_account_retriever doesn't try loading all banks on construction!